lib/pbench/test/unit/server/test_upload.py

```diff
@@ -679,8 +704,8 @@ def test_upload_invalid_metadata(
             "message": "at least one specified metadata key is invalid",
             "errors": [
                 "Metadata key 'server.archiveonly' value 'abc' for dataset must be a boolean",
-                "improper metadata syntax dataset.name=test must be 'k:v'",
-                "Key test.foo is invalid or isn't settable",
+                "Missing ':' terminator after 'dataset.name=test'",
```

Comment:
The terminator is actually supposed to be before the equals sign, right?  So, should that message say "in or after"?

Reply:
I'm not trying to do deep syntax analysis for errors. What it's *saying* is that it expects a ":" and didn't find one. The "sample case" here (which I didn't change) was someone hypothetically trying to use "=" as the separator, which isn't allowed. As a result it parsed the whole expression as a key, and failed because the terminator after the key isn't optional. I decided that, under the circumstances, this was good enough and didn't try to play with it more.